packages/button/src/ButtonBase.ts

```diff
@@ -220,18 +220,26 @@ export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
         }
     }
 
+    private isPendingState(): boolean {
+        return (
+            'pending' in this && (this as { pending: boolean }).pending === true
```

Comment:
duck-typing is fragile here and is not explicit. Two suggestions either you check for instance and cast only if instance is PendingState or you can add a clear comment saying that `pending` is a property injected here.